src/lib.rs

```diff
@@ -620,7 +624,29 @@ unsafe extern "C" fn start_game_clicked(
 
         let mut process = Command::new(state.game_path.clone());
 
-        process.spawn().ok();
+        let child = process.spawn().ok();
+        
+        #[cfg(unix)]
+        {
+            // For Unix systems only, spawn a new thread that waits for the process to exit.
+            // This avoids keeping the process in a zombie state and never letting go of it until
+            // the plugin is unloaded
+            thread::spawn(move || {
+                if child.is_none() {
+                    warn!("Failure starting the game process");
+                    return; 
+                };
```

Comment:
Two things:
1. Write this as `let Some(child) = child else { ... }` or `if let Some(child) = child { ... } ` or match, such that the unwrwap is not needed.
2. Do this outside the thread spawn, because there's no need to spawn a thread if it failed anyway (no need to do `.ok()` even, you can just match on the `Ok(...)` instead of the Some(...)`, you even get the error message to log then.